ngraph/core/reference/src/runtime/reference/depth_to_space.cpp

```diff
@@ -0,0 +1,116 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "ngraph/runtime/reference/depth_to_space.hpp"
+#include <cmath>
+#include <numeric>
+#include "ngraph/check.hpp"
+#include "ngraph/runtime/opt_kernel/reshape.hpp"
+
+namespace ngraph
+{
+    namespace runtime
+    {
+        namespace reference
+        {
+            void depth_to_space(const char* in,
+                                const Shape& in_shape,
+                                char* out,
+                                const Shape& out_shape,
+                                const std::size_t block_size,
+                                const op::DepthToSpace::DepthToSpaceMode mode,
+                                const size_t elem_size)
+            {
+                // DepthToSpace run in tree steps:
+                // - disperse data from depth channel
+                // - rearrange data so as appropriate chunks of data where close to their
+                //   destination place
+                // - squeeze data from respective dimensions
+                //
+                // First and third step doesn't change input data in memory, it change only the
+                // shape of input. From data layout perspective firs and third step may be
+                // omit. The second operation have to be perform on input data with dispared
+                // shape (x').
+                const size_t n_dim = in_shape.at(0);
+                const size_t c_dim = in_shape.at(1);
+                const size_t spatial_dim_index = 2;
+                const size_t spatial_dims = in_shape.size() - spatial_dim_index;
+                const auto c_dim_divider = static_cast<int>(std::pow(block_size, spatial_dims));
```

Comment:
Looks like `c_dim_divider` can have `size_t` type.

Reply:
switch to `size_t`